### PR TITLE
Adds fusion feature for Anserini

### PR DESCRIPTION
**Pull request by Stefan824**
The design is following :
- One TrecRun class that handles TrecRun-related methods including file-read, rescore, clone...
- One main class TrecRunFuser that calls methods from TrecRun and actually does the fusion
- One entry point class FuseTrecRuns that is in charge of only arguments and dispatches

Did a number of tests on codes but not thorough enough. Will test more if the design is accepted.

Should close https://github.com/castorini/anserini/pull/2355 and the results align with pyserini.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -516,5 +516,17 @@
...
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency> 
+    <dependency>
+        <groupId>org.junit.jupiter</groupId>
```
Comment (lintool) [inline pom.xml:526]: algin indentation with above

File: src/main/java/io/anserini/trectools/TrecRun.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  private Boolean reSort = false;
+  
+  // Constructor without reSort parameter
+  public TrecRun(Path filepath) throws IOException {
+    this(filepath, false);
+  }
+
+  // Constructor with reSort parameter
+  public TrecRun(Path filepath, Boolean reSort) throws IOException {
+      this.resetData();
```
Comment (lintool) [inline src/main/java/io/anserini/trectools/TrecRun.java:54]: consistent two space indentation?

File: src/main/java/io/anserini/trectools/TrecRun.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+      this.readRun(filepath);
+  }
+
+  // Constructor without parameters
+  public TrecRun() {
+      this.resetData();
+  }
+
+  private void resetData() {
+      runData = new ArrayList<>();
```
Comment (lintool) [inline src/main/java/io/anserini/trectools/TrecRun.java:66]: same here

File: src/test/resources/simple_trec_run_fusion_1.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+1 Q0 pyeb86on 1 24 reciprocal_rank_fusion_k=60
```
Comment (lintool) [inline src/test/resources/simple_trec_run_fusion_1.txt:1]: put all these files under a `fusion/` directory please.

File: src/test/java/io/anserini/trectools/TrecRunTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
+package io.anserini.trectools;
```
Comment (lintool) [inline src/test/java/io/anserini/trectools/TrecRunTest.java:1]: add boilerplate to here also?

**Review by lintool — CHANGES_REQUESTED** [review]
I don't think there's a need for a new package `src/main/java/io/anserini/trectools/` - just dump everything into `io.anserini.fusion` for now?

**Comment by lintool** [comment]
Oh, I see, you're modeling the structure after pyserini... yea, let's do a from-ground-up redesign.

File: src/test/resources/simple_trec_run_fusion_1.txt (hunk 4 above)
Comment (Stefan824) [reply src/test/resources/simple_trec_run_fusion_1.txt:1]: These test files were actually from Daniel early on; I didn't include my test files on the commit. Of course, I could move these to a `fusion/` directory or just delete them.

Comment (lintool) [reply src/test/resources/simple_trec_run_fusion_1.txt:1]: If these files are actually used, then please move. If they're not, then remove.

**Review by lintool — CHANGES_REQUESTED** [review]
Can you please look more carefully at the test cases?

File: src/test/java/io/anserini/fusion/TrecRunTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+public class TrecRunTest {
+  private TrecRun trecRun;
+  private Path sampleFilePath;
+
+  @Before
+  public void setUp() throws IOException {
+    sampleFilePath = Paths.get("runs/testlong/run.neuclir22-zh-en-splade.splade.topics.neuclir22-en.splade.original-desc_title.txt");
```
[the remark this answers is not in the document]
Comment (Stefan824) [reply src/test/java/io/anserini/fusion/TrecRunTest.java:37]: So sorry. The file seemed a bit too big for git. The plan was to bring this up when doing more tests after the code style/structure is accepted.

**Review by lintool — APPROVED** [review]
good enough for now...

**Pull request merged**